operator/gitops/argocd/pipeline-service/openshift-pipelines/chains-secrets-config.yaml

```diff
@@ -60,6 +60,10 @@ spec:
             - /bin/bash
             - -c
             - |
+              set -o errexit
+              set -o nounset
+              set -o pipefail
```

Comment:
I don't think it's a good idea to have these. 
For e.g. - we've statements in this script which are bound to fail in certain cases like -
The line SIG_KEY_DATA=$(kubectl get secret signing-secrets -n openshift-pipelines -o jsonpath='{.data}') might return a non-zero exit code if the secret doesn't exist. With set -o errexit, this will cause the entire script to fail.
This is why I think the CI is failing. 
Moreover, since we've `restartPolicy: OnFailure` on the job, it will lead to infinite loops of pod restarts.

Reply:
Setting these makes the script more robust, without them only the status of the last command is reported as the exit status. For commands that can return non zero status proper error handling should be implemented.

Reply:
Yes, agreed.

In that case, we should consider handling the exit codes, for .eg. -
`SIG_KEY_DATA=$(kubectl get secret signing-secrets -n openshift-pipelines -o jsonpath='{.data}') || true`

Or the other option suggested above: Since the restartPolicy is `OnFailure`, we handle this logic in the script itself  and exit with a status code of 0 to prevent unnecessary restarts which is how things are done currently.

Btw I've another PR raised which should fix `public-key` being empty issue - https://github.com/openshift-pipelines/pipeline-service/pull/776

Reply:
@ramessesii2
1. I've refactored the script so that `SIG_KEY_DATA` is not an issue anymore.
2. Do you think we need to change the restartPolicy? `OnFailure` makes sense to me as we would try to setup the secret until we can (e.g. in the event some RBAC that are needed aren't applied on the first try).

Reply:
Thanks Romain for the refactoring. 
`SIG_KEY_DATA` value was just an example. But we're good since the [default number of container restarts in case of back-off failure due to logical errors has a limit of 6](https://kubernetes.io/docs/concepts/workloads/controllers/job/#pod-backoff-failure-policy). I didn't notice this before due to exponential back-off delay. 
So, yes I agree with continuing with an `OnFailure` restartPolicy.